File: app/src/main/java/com/example/a39092/rxjavatest/module/main/view/MainActivity.java
package com.example.a39092.rxjavatest.module.main.view;

import android.os.Bundle;
import android.widget.ListView;

import com.example.a39092.rxjavatest.R;
import com.example.a39092.rxjavatest.model.bean.ResponseBody;
import com.example.a39092.rxjavatest.module.base.BaseActivity;
import com.example.a39092.rxjavatest.module.main.presenter.MainPresenter;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.reactivex.Observable;
import io.reactivex.Observer;
import io.reactivex.disposables.Disposable;

public class MainActivity extends BaseActivity implements MainView {


    @BindView(R.id.lv_movie)
    ListView lvMovie;


    private MainPresenter presenter;

    private LvMovieAdapter adapter;
    private List<ResponseBody.SubjectsBean> movies;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        initPresenter();
        initView();
        initData();
    }

    @Override
    public void initPresenter() {
        presenter = new MainPresenter(this);
    }

    @Override
    public void initView() {
        movies = new ArrayList<>();
        adapter = new LvMovieAdapter(movies, this);
        lvMovie.setAdapter(adapter);
        adapter.notifyDataSetChanged();
    }

    @Override
    public void initData() {
        presenter.getMovies(1, 10);
    }

    @Override
    public void onLoadMovies(final Observable<ResponseBody> response) {
        response.compose(this.<ResponseBody>bindToLifecycle()).subscribe(new Observer<ResponseBody>() {
            @Override
            public void onSubscribe(Disposable d) {
                showProgress();
            }

            @Override
            public void onNext(ResponseBody value) {
                movies.addAll(value.getSubjects());
            }

            @Override
            public void onError(Throwable e) {
            }

            @Override
            public void onComplete() {
                hideProgress();
                adapter.notifyDataSetChanged();
            }
        });
    }
}

File: app/src/main/java/com/example/a39092/rxjavatest/module/main/view/MainActivity.kt
package com.example.a39092.rxjavatest.module.main.view

import android.os.Bundle
import android.widget.ListView
import android.widget.Toast
import com.example.a39092.rxjavatest.R
import com.example.a39092.rxjavatest.model.bean.ResponseBody
import com.example.a39092.rxjavatest.module.base.BaseActivity
import com.example.a39092.rxjavatest.module.main.presenter.MainPresenter
import io.reactivex.Observable
import io.reactivex.rxkotlin.subscribeBy
import java.util.*

class MainActivity : BaseActivity(), MainView {


    var lvMovie: ListView? = null
    private var presenter: MainPresenter? = null

    private var adapter: LvMovieAdapter? = null
    private var movies: ArrayList<ResponseBody.SubjectsBean>? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initPresenter()
        initView()
        initData()
    }

    override fun initPresenter() {
        presenter = MainPresenter(this)
    }

    override fun initView() {
        lvMovie = findViewById(R.id.lv_movie) as ListView?
        movies = ArrayList<ResponseBody.SubjectsBean>()
        adapter = LvMovieAdapter(movies!!, this)
        lvMovie!!.adapter = adapter
        adapter!!.notifyDataSetChanged()
    }

    override fun initData() {
        presenter!!.getMovies(1, 10)
    }

    override fun onLoadMovies(response: Observable<ResponseBody>) {
        response.compose(this.bindToLifecycle()).doOnSubscribe { _ ->
            showProgress()
        }.subscribeBy(
                onNext = { responseBody -> movies!!.addAll(responseBody.subjects!!) },
                onError = { e -> Toast.makeText(this, e.toString(), Toast.LENGTH_SHORT).show() },
                onComplete = {
                    hideProgress()
                    adapter!!.notifyDataSetChanged()
                })
    }
}
